packages/block-library/src/social-link/edit.native.js

```diff
@@ -0,0 +1,192 @@
+/**
+ * External dependencies
+ */
+import { View, Animated, Easing, TouchableWithoutFeedback } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import {
+	ToolbarGroup,
+	ToolbarButton,
+	LinkSettings,
+} from '@wordpress/components';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { __, sprintf } from '@wordpress/i18n';
+import { link, Icon } from '@wordpress/icons';
+/**
+ * Internal dependencies
+ */
+import { getIconBySite, getNameBySite } from './social-list';
+import styles from './editor.scss';
+
+const ANIMATION_DELAY = 300;
+const ANIMATION_DURATION = 400;
+
+const linkSettingsOptions = {
+	url: {
+		label: __( 'URL' ),
+		placeholder: __( 'Add URL' ),
+		autoFocus: true,
+	},
+	linkLabel: {
+		label: __( 'Link label' ),
+		placeholder: __( 'None' ),
+	},
+	footer: {
+		label: __( 'Briefly describe the link to help screen reader user' ),
+	},
+};
+
+const SocialLinkEdit = ( {
+	attributes,
+	setAttributes,
+	isSelected,
+	onFocus,
+} ) => {
+	const { url, service } = attributes;
+	const [ isLinkSheetVisible, setIsLinkSheetVisible ] = useState( false );
+	const [ isValue, setIsValue ] = useState( !! url );
```

Comment:
It took me a minute to understand what this meant 😅, I'm usually not picky with names but what do you think if we change it to something like `hasUrl` or `isUrlSet` or something related to the `url`?

Reply:
It makes sense!